src/AppInstallerCommonCore/AppInstallerLogging.cpp

```diff
@@ -143,6 +157,72 @@ namespace AppInstaller::Logging
     {
         return out << std::hex << std::setw(8) << std::setfill('0');
     }
+
+    DiagnosticLogger& ThreadGlobals::GetDiagnosticLogger()
+    {
+        if (!m_pDiagnosticLogger)
+        {
+            try
+            {
+                std::call_once(diagLoggerInitOnceFlag, [this]()
+                {
+                    InitDiagnosticLogger();
+                });
+            }
+            catch (...)
+            {
+                // May throw std::system_error if any condition prevents calls to call_once from executing as specified
+                // Loggers are best effort and shouldn't block core functionality. So eat up the exceptions here
```

Comment:
I'm not sure about this one; if there is something blocking all logging we probably want to know.

Reply:
Agree. I cab log this failure in the text file.